ledger/byron/byron.go

```diff
@@ -731,6 +742,17 @@ func (h *ByronEpochBoundaryBlockHeader) Era() common.Era {
 	return EraByron
 }
 
+func (h *ByronEpochBoundaryBlockHeader) BlockBodyHash() common.Blake2b256 {
+	// BodyProof is guaranteed to be common.Blake2b256 for valid Byron blocks
+	// as per the Cardano Byron era specification
+	if bodyProof, ok := h.BodyProof.(common.Blake2b256); ok {
+		return bodyProof
+	}
```

Comment:
I don't think this is actually going to work. The `BodyProof` field is type `any`, the CBOR decoder won't use `common.Blake2b256` on its own in this case, and we do not explicitly assign to that field anywhere.